api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseDao.java

```diff
@@ -385,6 +385,20 @@ protected boolean containsAlias(Iterator<CriteriaImpl.Subcriteria> subcriteriaIt
 		    handleDate(propertyName, dateRangeParam.getUpperBound())))));
 	}
 	
+	//method for dateRanges that dont rely on a single property like TestOrders
+	protected Optional<Criterion> handleDateRange(String scheduledDate, String dateActivated, String dateStopped,
+	        String autoExpireDate, DateRangeParam dateRangeParam) {
+		if (dateRangeParam == null) {
+			return Optional.empty();
+		}
+		
+		return Optional.of(and(toCriteriaArray(Stream.of(
+		    Optional.of(or(toCriteriaArray(Stream.of(handleDate(scheduledDate, dateRangeParam.getLowerBound()),
+		        handleDate(dateActivated, dateRangeParam.getLowerBound()))))),
+		    Optional.of(or(toCriteriaArray(Stream.of(handleDate(dateStopped, dateRangeParam.getUpperBound()),
+		        handleDate(autoExpireDate, dateRangeParam.getUpperBound())))))))));
+	}
+	
```

Comment:
Thanks for this! This actually quite clever and solves the problem well. However, I think this method rightly belongs in the `FhirServiceRequestDaoImpl` file rather than here, since it's pretty specific to the implementation of `ServiceRequest`. The functions in `BaseDao` are largely intended to be used in more than one place.